presto-main/src/test/java/io/prestosql/type/TestDateTimeOperatorsBase.java

```diff
@@ -251,6 +252,13 @@ public void testIsDistinctFrom()
         assertFunction("DATE '2013-10-27' IS DISTINCT FROM NULL", BOOLEAN, true);
     }
 
+    @Test
+    public void testDateCastFromVarchar()
+    {
+        assertFunction("DATE '2013-02-02'", DATE, toDate(new DateTime(2013, 2, 2, 0, 0, 0, 0, UTC)));
+        assertInvalidFunction("DATE '5881580-07-12'", INVALID_CAST_ARGUMENT, "Value cannot be cast to date: 5881580-07-12");
+    }
```

Comment:
Can you also include a test case when value fails within `DATE_FORMATTER.parseMillis(value)` method (due to overflow)?

Reply:
have applied it